### PR TITLE
BranchIndexingTrigger support

### PR DESCRIPTION
**Pull request by offa**
Support _BranchIndexingTrigger_ and set values accordingly. 

Builds triggered by Branch Indexing log an "unspported cause type(s)" warning currently.

As `BranchEventCause`, `BranchIndexingCause` is only relevant if the optional _branch-api_ plugin is present.

-----------------------

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by StefanSpieker — APPROVED** [review]

**Review by fabiodcasilva — APPROVED** [review]

**Comment by Schleuse** [comment]
May I ask a question about this?

Before this got merged `env.BUILD_USER_ID`, `env.BUILD_USER` & `env.BUILD_USER_EMAIL` were populated with details about the committer that caused the branch indexing. Now the variables are always populated with `branchIndexing` regardless of the user triggering it.

How do you get the information about the user that caused the branch indexing?

**Comment by offa** [comment]
Interesting, where did these values come from? `BranchIndexingTrigger` wasn't handled previously, thus only a warning was issued, but no values were set.

**Comment by Schleuse** [comment]
Sorry, my bad! I've got a bit lost in our Pipeline :stuck_out_tongue: 

We've used `build-user-vars-plugin` to extract the user information for builds that got triggered manual by the jenkins ui. When no `env.BUILD_USER_ID` could be determined we've used git to extract the last committer. 

Now `env.BUILD_USER_ID` is also filled when the build got triggered by branch indexing and the extraction of the user data from git gets skipped.

```groovy
wrap([$class: 'BuildUser']) {
  config.user = [
    id : env.BUILD_USER_ID,
    name : env.BUILD_USER,
    email : env.BUILD_USER_EMAIL
  ]
}

if (config.user.id == null) {
  config.user = [
    id : sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%an'").trim(),
    email : sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%ae'").trim()
  ]
}
```

Thanks for the fast reply :duck:

**Pull request merged**
